engines/wintermute/base/gfx/opengl/meshx.cpp

```diff
@@ -302,7 +302,87 @@ bool MeshX::updateShadowVol(ShadowVolume *shadow, Math::Matrix4 &modelMat, const
 		return false;
 	}
 
-	return shadow->addMesh(_adjacency.data(), modelMat, light, extrusionDepth);
```

Comment:
What's the reason for moving this here?

Reply:
This way the ShadowVolume class does not need to have a function with some mesh parameters (vertex buffer, index buffer, adjacency information). Instead, it just offers a function to add a vertex and the actual procedure is implemented by the particular mesh class, which has access to the aforementioned state anyways.